framework/src/media/audio/audio_manager.c

```diff
@@ -682,17 +632,26 @@ static audio_manager_result_t set_audio_volume(audio_io_direction_t direct, uint
 
 audio_manager_result_t audio_manager_init(void)
 {
-	return AUDIO_MANAGER_SUCCESS;
-}
+	static int am_initialized = 0;
+	int card_id;
 
-audio_manager_result_t init_audio_stream_in(void)
-{
-	return init_audio_stream(INPUT);
-}
+	if (am_initialized) {
+		return AUDIO_MANAGER_SUCCESS;
+	}
 
-audio_manager_result_t init_audio_stream_out(void)
-{
-	return init_audio_stream(OUTPUT);
+	am_initialized = 1;
+
+	for (card_id = 0; card_id < CONFIG_AUDIO_MAX_INPUT_CARD_NUM; card_id++) {
+		pthread_mutex_init(&(g_audio_in_cards[card_id].card_mutex), NULL);
+	}
+	find_audio_card(INPUT);
+
+	for (card_id = 0; card_id < CONFIG_AUDIO_MAX_OUTPUT_CARD_NUM; card_id++) {
+		pthread_mutex_init(&(g_audio_out_cards[card_id].card_mutex), NULL);
+	}
+	find_audio_card(OUTPUT);
+
+	return AUDIO_MANAGER_SUCCESS;
 }
 
```

Comment:
it it okay to return AUDIO_MANAGER_SUCCESS on  find_audio_card() returns AUDIO_MANAGER_NO_VALID_CARD cases?

Reply:
Fixed it